### PR TITLE
Add Kiva deep ground depth check and recalculation if needed

### LINKED ISSUE #8859: Automatic deep ground depth in Kiva not being set correctly
Issue overview
--------------
There is a bug where the automatic deep ground depth for shallow water tables isn't getting set to 1m below the bottom of the foundation like it should. You can get around this in your file by explicitly defining the deep ground depth in Foundation:Kiva:Settings.  In the defect file, there are no "blocks", so the deep ground boundary is never adjusted.  Before going into [this loop](https://github.com/NREL/EnergyPlus/blob/ccad7ddb99134ce8d6901fea16232d35d5632a18/src/EnergyPlus/HeatBalanceKivaManager.cc#L983), we need to check for cases where the deep ground is shallower than the foundationDepth + wallDepthBelowGrade:

image

[Shoebox.zip](https://github.com/NREL/EnergyPlus/files/6750806/Shoebox.zip)

### Details
Some additional details for this issue (if relevant):
 - EnergyPlus v9.4
 - Unmethours link: [https://unmethours.com/question/55972/energyplus-kiva-error/](https://unmethours.com/question/55972/energyplus-kiva-error/)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by matthew-larson**
Pull request overview
---------------------
 - Fixes #8859 
 - This pull request moves the section of code that checks and calculates `fnd.deepGroundDepth` to an inline function and adds a check that the deep ground depth is below the total depth of wall below grade.  If the deep ground depth is not below the total depth of wall below grade, the deep ground depth is modified to be 1 meter below the total depth of wall below grade and a warning is generated stating the deep ground depth was modified.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 
### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts

### CODE REVIEW

File: src/EnergyPlus/HeatBalanceKivaManager.cc [hunk 1]
```diff
@@ -1119,6 +1108,27 @@ bool KivaManager::setupKivaInstances(EnergyPlusData &state)
     return ErrorsFound;
 }
 
+Real64 KivaManager::getDeepGroundDepth(Kiva::Foundation fnd)
+{
+    Real64 totalDepthOfWallBelowGrade = fnd.wall.depthBelowSlab + (fnd.foundationDepth - fnd.wall.heightAboveGrade) + fnd.slab.totalWidth();
+    if (fnd.deepGroundDepth < totalDepthOfWallBelowGrade + 1.0) {
+        fnd.deepGroundDepth = totalDepthOfWallBelowGrade + 1.0;
+    }
```
Comment (mitchute) [inline src/EnergyPlus/HeatBalanceKivaManager.cc:1116]: This is the actual bug fix. Looks OK to me.

**Review by mitchute — APPROVED** [review]
This looks good. I'm going to test this on Linux and Mac, and then probably just merge it. I can't imagine this causing issues if those work locally.

**Comment by mitchute** [comment]
OK, built and tested locally on Linux and Mac. I'm not going to bother CI with this one. Merging.

**Comment by matthew-larson** [comment]
@mitchute Thanks!

**Pull request merged**